app.rb

```diff
@@ -0,0 +1,167 @@
+    print 'specialization : '
+    specialization = gets.chomp
+
+    new_teacher = Teacher.new(age, specialization, name)
+    @people_list << new_teacher
+    puts 'Teacher created successfully'
+    puts "\n"
+  end
+
+  def create_student
+    puts "\n"
+    puts 'Please provide these information :'
+    print 'Classroom (number) :'
+    classroom = gets.chomp
+    print 'Age : '
+    age = gets.chomp
+    print 'Name : '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N] : '
+    parent_permission = gets.chomp
+    permission = true
+    permission = false if parent_permission.upcase == 'N'
+    new_student = Student.new(age, classroom, name, parent_permission: permission)
+    @people_list << new_student
+    puts 'Student created successfully'
+    puts "\n"
+  end
+
+  def create_person
+    puts "\n"
+    print 'Do you want to create a Student (1) or a Teacher (2) [Input the number]: '
+    choice = gets.chomp
+    if choice == '1'
+      create_student
+    else
+      create_teacher
+    end
+  end
+
+  def create_book
+    puts "\n"
+    puts 'Please provide these information :'
+    print 'Title: '
+    title = gets.chomp
+    print 'Author : '
+    author = gets.chomp
+    new_book = Book.new(title, author)
+    @books_list << new_book
+    puts 'Book created successfully'
+    puts "\n"
+  end
+
+  def create_rental
+    puts "\n"
+    if @books_list.empty? || @people_list.empty?
+      puts 'Please make sure you have books and persons'
+    else
+      puts 'Select a book from the following list by number'
+      list_all_books
+      book_position = gets.chomp.to_i
+      puts 'Select a person from the following list by number (not id)'
+      list_all_people
+      person_position = gets.chomp.to_i
+      print 'Date (2023/04/13): '
+      date = gets.chomp
+      new_rent = Rental.new(date, @books_list[book_position], @people_list[person_position])
+      @rentals_list << new_rent
+      puts 'Rental created successfully'
+    end
+    puts "\n"
+  end
+
+  def display_menu
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all book'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def menu()
+    choice = ''
+    while choice != 7.to_s
+      display_menu
+      choice = gets.chomp
+      case choice
+      when '1' then list_all_books
+      when '2' then list_all_people
+      when '3' then create_person
+      when '4' then create_book
+      when '5' then create_rental
+      when '6'then list_all_rental
+      when '7'then exit_app
+      else
+        puts "Please select a number between 1 and 7\n\n"
+      end
+    end
+  end
+  # rubocop:enable Metrics/CyclomaticComplexity
+
+  def exit_app
+    puts 'Thank you for using this app!'
+    exit
+  end
```

Comment:
Well done so far. Your project is looking nice. However, as part of the project requirements, you are supposed to define the entry point, this will be a method called main that is invoked at the end of your main.rb file. It should do the following: 

1. Present the user with a list of options to perform.
2. Lets users choose an option.
3. If needed, ask for parameters for the option.
4. Have a way to quit the app

All the above listed points are to be done in the main.rb file and not the app.rb file as you have done here. Kindly check and move them in order to match the project requirements as shown :point_down: 


![Screenshot from 2023-04-14 13-11-09](https://user-images.githubusercontent.com/65233860/232054407-c01614c2-4927-4f10-a551-a3dc95e0549c.png)